### PR TITLE
wip: attempt to simplify workflows handling

### PR DESCRIPTION
**Pull request by metacosm**
- **refactor: remove now unneeded NameSetter interface**
- **refactor: simplify workflow handling**
- **refactor: initial work to combine ManagedWorkflow and regular one**

### CODE REVIEW

File: operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/workflow/DefaultManagedWorkflow.java [hunk 1]
```diff
@@ -10,37 +9,25 @@
...
+    final var alreadyResolved = new HashMap<String, DependentResourceNode>(orderedSpecs.size());
+    for (DependentResourceSpec spec : orderedSpecs) {
+      final var node = new UnresolvedDependentResourceNode(spec);
+      alreadyResolved.put(spec.getName(), node);
+      spec.getDependsOn()
+          .forEach(depend -> node.addDependsOnRelation(alreadyResolved.get(depend)));
     }
+
+    this.workflow = new DefaultWorkflow<>(alreadyResolved.values(), false, hasCleaner);
+    this.orderedSpecs = orderedSpecs; // todo: remove
```
Comment (xstefank) [inline operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/workflow/DefaultManagedWorkflow.java:30]: I don't understand these todos in this class. Do you still plan to work on them in this PR?

**Review by xstefank — COMMENTED** [review]
Other than todos I don't see anything wrong.

**Comment by csviri** [comment]
@metacosm I would prefer to review this together, I fail to see how this simplifies things.

**Comment by csviri** [comment]
Might be also better to do such refactorings step by step with limited scoped PR- if possible.

**Comment by metacosm** [comment]
This isn't done yet so, indeed, this doesn't really simplify things at the moment… also, I'm not sure the individual steps will make any difference since they will only make sense once the whole is completed (if it can actually be achieved as expected, which I'm not yet sure of as this is still rather exploratory).

**Comment by metacosm** [comment]
Replaced by #2566

**Pull request closed without merging**